##This patch sets libraries' installation paths according to the Fedora packaging guidelines
--- a/src/bindings/ruby/CMakeLists.orig.txt	2014-11-26 13:31:14.000000000 +0100
+++ b/src/bindings/ruby/CMakeLists.txt	2014-12-28 23:25:47.035666311 +0100
@@ -177,7 +177,7 @@
 if (UNIX OR CYGWIN) 
   execute_process(COMMAND "${RUBY_EXECUTABLE}" -e "print RUBY_PLATFORM"
     OUTPUT_VARIABLE RUBY_PLATFORM)
-  set(RUBY_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/ruby/site_ruby/${RUBY_VERSION_MAJOR}.${RUBY_VERSION_MINOR}/${RUBY_PLATFORM})
+  set(RUBY_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/ruby/vendor_ruby)
 else()
   set(RUBY_PACKAGE_INSTALL_DIR ${MISC_PREFIX}bindings/ruby)
 endif()
--- a/src/bindings/csharp/CMakeLists.orig.txt	2014-11-26 13:31:17.000000000 +0100
+++ b/src/bindings/csharp/CMakeLists.txt	2014-12-29 11:37:43.839593422 +0100
@@ -213,7 +213,7 @@
 #
 set(CSHAPR_PACKAGE_INSTALL_DIR)
 if (UNIX OR CYGWIN)
-  set(CSHAPR_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/mono/libsbmlcsP)
+  set(CSHAPR_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/lib/mono/libsbmlcsP)
 else()
   set(CSHAPR_PACKAGE_INSTALL_DIR ${MISC_PREFIX}bindings/csharp)
 endif()
--- a/src/bindings/octave/CMakeLists.orig.txt	2014-11-26 13:31:10.000000000 +0100
+++ b/src/bindings/octave/CMakeLists.txt	2014-12-29 11:44:51.031069073 +0100
@@ -67,7 +67,7 @@
 if (UNIX OR CYGWIN) 
 execute_process(COMMAND "${OCTAVE_CONFIG_EXECUTABLE}" -p CANONICAL_HOST_TYPE
     OUTPUT_VARIABLE OCTAVE_PLATFORM OUTPUT_STRIP_TRAILING_WHITESPACE)
-  set(OCTAVE_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/octave/site/oct/${OCTAVE_PLATFORM})
+  set(OCTAVE_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/octave/packages/SBML-${PACKAGE_VERSION})
 else()
   set(OCTAVE_PACKAGE_INSTALL_DIR ${MISC_PREFIX}bindings/octave)
 endif()
--- a/src/bindings/perl/CMakeLists.orig.txt	2014-11-26 13:31:13.000000000 +0100
+++ b/src/bindings/perl/CMakeLists.txt	2014-12-29 11:48:53.705649453 +0100
@@ -232,7 +232,7 @@
   string(REPLACE "'" "" PERL_PLATFORM ${PERL_PLATFORM})
   string(REPLACE ";" "" PERL_PLATFORM ${PERL_PLATFORM})
   string(REPLACE "archname=" "" PERL_PLATFORM ${PERL_PLATFORM})
-  set(PERL_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/perl5/site_perl/${PERL_VERSION}/${PERL_PLATFORM})
+  set(PERL_PACKAGE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/perl5/vendor_perl)
   set(PERL_PACKAGE_INSTALL_BIN_DIR ${PERL_PACKAGE_INSTALL_DIR}/auto/libSBML)
 else()
   set(PERL_PACKAGE_INSTALL_DIR ${MISC_PREFIX}bindings/perl)

